### PR TITLE
Off Topic Names

### PR DESCRIPTION
**Pull request by Shivansh-007**
Tada

### CODE REVIEW

**Comment by brtwrst** [comment]
I'm not sure why we need an off topic names cog and you give no explanation here as to what it's for.
Was that a requested feature?
Was it really necessary to cram this paginator thing into this as well and convert it to a package?
I see this changes a bunch of hard coded relative paths but leaves some others as they are.
Did you check if all the current cogs still work with this?

I can't test it right now as I'm busy with work and I'm not really keen on double checking if all current cogs still work with this.

**Comment by Shivansh-007** [comment]
1. [Context on Discord](https://discord.com/channels/473161189120147456/769439015094059019/819847447232118834).
2. Above ^
3. Yes it is needed to convert into a package so that we can import `pagination` and `OffTopicConverter`, otherwise there are import errors.
4. Yes, there is only change in the `.json/.log` files, and they work perfectly fine, there are no other changes.

**Comment by brtwrst** [comment]
I can promise you I know of at least one other cog that uses relative paths and will definitely not work any more and you tell me "they work perfectly fine".

Please - just revert back the module stuff and make your stuff work with the current setup.

**Comment by Shivansh-007** [comment]
Changes mentioned @brtwrst!

File: python/bot.py [hunk 1]
```diff
@@ -55,6 +55,17 @@ def user_is_ignored(self, user):
             return True
         return False
 
+    def user_is_hero(self, user):
+        try:
+            user_roles = [role.id for role in user.roles]
+        except AttributeError:
+            return False
+        community_hero_role = self.config['admin_roles']
```
Comment (brtwrst) [inline python/bot.py:63]: should be hero_role?

File: python/cogs/utils/converters.py [hunk 2]
```diff
@@ -0,0 +1,37 @@
...
+from discord.ext.commands import BadArgument, Converter
+
+ALLOWED_CHARACTERS = "ABCDEFGHIJKLMNOPQRSTUVWXYZ!?'`-"
+UNICODE_CHARACTERS = '𝖠𝖡𝖢𝖣𝖤𝖥𝖦𝖧𝖨𝖩𝖪𝖫𝖬𝖭𝖮𝖯𝖰𝖱𝖲𝖳𝖴𝖵𝖶𝖷𝖸𝖹ǃ？’’-'
+
+
+class OffTopicName(Converter):
+    """A converter that ensures an added off-topic name is valid."""
+
+    @classmethod
```
Comment (brtwrst) [inline python/cogs/utils/converters.py:12]: why is this a classmethod?

File: python/bot.py (hunk 1 above)
Comment (Shivansh-007) [reply python/bot.py:63]: Yes, copied the function from `user_is_admin` and forgot to change the config role.

File: python/cogs/utils/converters.py (hunk 2 above)
Comment (Shivansh-007) [reply python/cogs/utils/converters.py:12]: Right, it was using the class attribute `ALLOWED_CHARACTERS` earlier, but not i shifted that to a constant, which is unnecessary, will fix it! Thanks.

**Comment by Shivansh-007** [comment]
I don't think I am willing to contribute to felix anymore, so I am closing this, if anyone wants to work on this they can continue from where I left.

Thanks!

**Pull request closed without merging**